src/middleware.js

```diff
@@ -7,10 +7,14 @@ import type { RouterAction } from './types';
 
 import {
   PUSH, REPLACE, GO,
-  GO_BACK, GO_FORWARD
+  GO_BACK, GO_FORWARD,
+  BLOCK, UNBLOCK
 } from './types';
 
 type MiddlewareArgs = { history: History };
+
+let unblock;
```

Comment:
SINGLETON STATE :rage3: 😛 

I'd scope this to the history, i.e. move it to line 19 like

```javascript
({ history }): MiddlewareArgs) => {
  let unblock;
  () => ...
```